### DISCUSSION TITLE
Confirm before deleting branch!!

### DISCUSSION THREAD

**Discussion by sky-xo** [planetscale/discussion#636] [Feature Requests] [answered]
I had just finished a deploy request a couple hours ago from my "dev" branch to my "main". On the deploy request branch there was a sentence that said something like "This branch can be deleted <delete link>". I instinctively pressed it (since that's what i do after merging github pull requests). However that dev branch had lots of data on it that our developers were testing out, and now it's all gone!!! I can't even access it anywhere in my backups (from what I can tell).

Is there a way to access this deleted branch? At the VERY LEAST you should warn me and make me confirm before deleting, or provide an undo! This is insannnnity!! Can't believe we lost all of our dev data without so much of a confirmation or warning of any sort.

**Comment by mscoutermarsh** [comment]
Hi @skyriverbend,

Could you please email support? We can check if there is a backup available for the branch.

**Comment by mscoutermarsh** [comment, marked as answer]
This was resolved via support. We have also made a change to the backups page which will show all recently deleted branch backups, making it easy to restore them.

The button is also being updated to include a confirm step.

**Comment by sky-xo** [comment]
Wow, what a speedy and professional response. Solving the issue and immediately making changes in the app to stop this from happening in the future. You definitely won my trust. 🙏 ❤️ ✨
